src/world/Ball.cpp

```diff
@@ -16,138 +13,72 @@ namespace world {
 bool Ball::exists = false;
 
 Ball::Ball()
-        :pos(Vector2()), vel(Vector2()), filteredVel(Vector2()), visible(false) { }
+        : expectedPosition(Vector2()), expectedVelocity(Vector2()), filteredVelocity(Vector2()),
+          visibleByAnyCamera(false) {
+}
 
 Ball::Ball(const roboteam_proto::WorldBall &copy)
-        :pos(copy.pos()), vel(copy.vel()), filteredVel(copy.vel()),
-         visible(copy.visible()) {
+        : expectedPosition(copy.pos()), expectedVelocity(copy.vel()), filteredVelocity(copy.vel()),
+          visibleByAnyCamera(copy.visible()) {
     exists = exists || copy.area() || Vector2(copy.pos()).isNotNaN();
     if (! exists) std::cout << "BallPtr message has existence = 0!!" << std::endl;
```

Comment:
This should be refactored (together with existance as a concept)

Reply:
For now it's ok, since this probably requires changes in roboteam_proto and roboteam_world too